plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -838,4 +839,20 @@ public void setColumnType(ConnectorSession session, JdbcTableHandle handle, Jdbc
     {
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support setting column types");
     }
+
+    @Override
+    public OptionalInt getMaxColumnNameLength(ConnectorSession session)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
```

Comment:
I was actually doing a small research around whether it's possible to query a database what's the alias name limit, but could not find anything that worked in real life. Did you manage to test/confirm that it does?

Reply:
Yes, it depends on exact db and driver being used. That's why the default impl. just returns "No limit" while in connectors where we can verify the limits we have an impl.

For the ones where i added an impl we do have tests which would fail if you change the impl or return "no limits".

Reply:
> That's why the default impl. just returns "No limit"

Can we have same implementation for such connectors too. As I understand JDBC standard in such case `connection.getMetaData().getMaxColumnNameLength();` should also say something reasonable like `Integer.MAX_VALUE`

Reply:
> connection.getMetaData().getMaxColumnNameLength(); should also say something reasonable like Integer.MAX_VALUE

No. The JDBC API says to return 0 for cases where there is no limit or the limit is unknown, no Integer.MAX_VALUE and databases seem to follow that instead of returning Integer.MAX_VALUE.

Reply:
I think this was just a discussion. no change done.

note that the database metadata approach needs to be explicitly used by a connector. If database metadata is known to be wrong for some database then they can provide their own override for `getMaxColumnNameLength`.